inductiva/tasks/task.py

```diff
@@ -174,13 +175,73 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
+    def kill(
+            self,
+            wait_timeout: Optional[Union[float,
+                                         int]] = None) -> Union[bool, None]:
+        """Tries to kill a task.
+        If wait_timeout is None, the function will send the kill
+        request, and return none. If wait_timeout is a positive number,
+        the function will send the kill request and wait (wait_timeout
+        seconds) for the task to be killed. We return True if the task
+        was successfully killed, and False otherwise.
+        Args:
+            wait_timeout: number of seconds to wait for the kill command
+            or none if we don't want to wait.
+        Returns:
+            None if we don't wait, True if the kill command was successful,
+            False if it was not.
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        if not isinstance(wait_timeout, (float, int, type(None))) or (
+                wait_timeout is not None and wait_timeout <= 0.0):
+            raise ValueError("Wait timeout must be a positive number or None.")
+
+        maxretries = constants.MAX_API_RETRIES
+
+        while maxretries > 0:
+            maxretries -= 1
+            try:
+                task_status = self.get_status()
+
+                if task_status in _TASK_TERMINAL_STATUSES:
+                    break
+
+                path_params = self._get_path_params()
+                self._api.kill_task(path_params=path_params)
+                break
+            except exceptions.ApiException as exc:
+                if maxretries == 0:
+                    raise RuntimeError(
+                        "Something went wrong while sending"
+                        "the kill command. Try again Later") from exc
+                time.sleep(1)
+
+        if wait_timeout is None:
+            return None
+
+        success = True
+        start = time.time()
+        status = self.get_status()
+
+        while status == models.TaskStatusCode.PENDINGKILL:
+            status = self.get_status()
```

Comment:
use the walrus operator to get the status in the if condition:

```suggestion
        while(status := self.get_status()) == models.TaskStatusCode.PENDINGKILL:
```